internal/parse/parser.go

```diff
@@ -187,57 +215,105 @@ func isNameByte(c byte) bool {
-	stringf  func(string) bool
-	stringf0 func() bool
-	result   []bool
-	input    string
-	data     []string
+// skipName returns false if the parser is not on a name. Otherwise it advances
+// the parser until it is on the first non name byte and returns true.
+func (p *Parser) skipName() bool {
+	if p.pos >= len(p.input) {
+		return false
+	}
+	start := p.pos
+	for p.pos < len(p.input) && isNameByte(p.input[p.pos]) {
+		p.pos++
+	}
+	return p.pos > start
 }
 
-func TestRunTable(t *testing.T) {
-	var p = NewParser()
-	var parseTests = []parseHelperTest{
+// These functions attempt to parse some construct, they return a bool and that
+// construct, if they n't parse they return false, restore the parser and leave
+// the default value in  other return type
 
-		{bytef: p.peekByte, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.peekByte, result: []bool{false}, input: "b", data: []string{"a"}},
-		{bytef: p.peekByte, result: []bool{true}, input: "a", data: []string{"a"}},
+func (p *Parser) parseIdentifier() (string, bool) {
+	if p.pos >= len(p.input) {
+		return "", false
+	}
+	if p.skipByte('*') {
+		return "*", true
+	}
 
-		{bytef: p.skipByte, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.skipByte, result: []bool{false}, input: "abc", data: []string{"b"}},
-		{bytef: p.skipByte, result: []bool{true, true}, input: "abc", data: []string{"a", "b"}},
+	idStart := p.pos
+	if p.skipName() {
+		return p.input[idStart:p.pos], true
+	}
+	return "", false
+}
 
-		{bytef: p.skipByteFind, result: []bool{false}, input: "", data: []string{"a"}},
-		{bytef: p.skipByteFind, result: []bool{false, true, true}, input: "abcde", data: []string{"x", "b", "c"}},
-		{bytef: p.skipByteFind, result: []bool{true, false}, input: "abcde ", data: []string{" ", " "}},
+// parseFullName parses a column name or Go field name, optionally dot-prefixed by
+// its table or type name respectively.
+func (p *Parser) parseFullName(idc idClass) (FullName, bool) {
+	cp := p.save()
+	var fn FullName
+	p.skipSpaces()
+	if id, ok := p.parseIdentifier(); ok {
+		fn.Prefix = id
+		if p.skipByte('.') {
+			if id, ok := p.parseIdentifier(); ok {
+				fn.Name = id
+				return fn, true
+			}
+		} else {
+			// A column name specified without a table prefix is a name not a
+			// prefix
+			if idc == columnId {
+				fn.Name = fn.Prefix
+				fn.Prefix = ""
+			}
+			return fn, true
+		}
+	}
+	cp.restore()
+	return fn, false
+}
 
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "abc    d", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "     abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{true}, input: "  \t  abcd", data: []string{}},
-		{stringf0: p.skipSpaces, result: []bool{false}, input: "\t  abcd", data: []string{}},
+// parseInputExpression parses an SDL input go-defined type to be used as a
+// query argument.
```

Comment:
This sentence is reading a bit curiously. How about:
```
// parseInputExpression parses an input expression of the form $Type.name.
```

Reply:
changed.